Explain how to add a search engine to Vanadium and set as default

kekkoudesu opened the pull request:
It isn't obvious that Vanadium supports adding search engines, which leads users to open issues asking for this feature: https://github.com/GrapheneOS/Vanadium/issues/254

A short note explaining how to add a search engine and set it as the default search engine will help new users setup their new browser with their choice of privacy-respecting search engine.

kekkoudesu committed:
Explain how to add a search engine to Vanadium and set as default

It isn't obvious that custom search engines are already supported in Vanadium, which causes users to open issues asking for this feature: https://github.com/GrapheneOS/Vanadium/issues/254

A short note explaining how to add a search engine and set it as the default search engine will help new users setup their new browser with their choice of privacy-respecting search engine.

jahway603 commented:
@thestinger I'm following the process you've outlined in this PR and it does not work with Startpage.com, but I also followed this same process in Bromite with Startpage.com and it did work. Should I open a new issue?

thestinger commented:
Please use Matrix or the discussion forum to ask questions and don't specifically direct questions towards me.

thestinger commented:
Search engines not being compatible with mobile Chromium is not Vanadium specific and not a bug. They choose to do things in a way that's not compatible including using POST instead of GET for searches.

kekkoudesu commented:
@jahway603 
> I'm following the process you've outlined in this PR and it does not work with Startpage.com, but I also followed this same process in Bromite with Startpage.com and it did work. Should I open a new issue?

I'm the author of this PR. As explained previously, the reason Chromium-based browsers on Android like Vanadium don't pick up the OpenSearch document served by Startpage is because they have chosen to use POST HTTP requests by default for privacy reasons (it doesn't show your searches in your browsing history), which don't expose the query parameter used for making searches in the URL. You can change the HTTP Request Method to POST here: https://www.startpage.com/do/settings

Once you do that, it will pick up the search engine properly. Additionally, you will need to do this in normal browsing mode, not Incognito Mode. Any future searches with Startpage will be made through the GET method. To replicate the benefit of not storing searches in history for Chromium-based browsers on Android like Vanadium, the only way to do this is to always use Incognito mode or regularly clear your history manually.

It would be less confusing if Chromium-based browsers allowed you to to simply add the search engine when an OpenSearch document is served as Firefox on desktop allows you to do as soon as you land on the website. You may wish to open a feature request about this on bugs.chromium.org, because while these browsers are properly parsing the OpenSearch document in most circumstances, they apparently decide to ignore it if the resulting search uses a POST method.

Chromium-based browsers on Android may allow you to add search engines manually sometime this year, if all goes well, so adding search engines may be less confusing in the future: https://bugs.chromium.org/p/chromium/issues/detail?id=1440189

kekkoudesu committed:
Clarify that HTTP GET Requests must be used for the OpenSearch document to be parsed

Chromium-based browsers on Android only parse the OpenSearch document if GET HTTP requests are used with the query. Even if the website serves an OpenSearch document, and the user performs a search on the site, it will not be detected by the browser if it uses a POST HTTP request to handle the query.

Startpage is one example of a search engine which properly supports the OpenSearch standard, but uses POST HTTP requests to prevent searches showing up in the user's browsing history, and so Chromium-based browsers ignore the OpenSearch document.
